sdk/tables/aztables/tableClient.go

```diff
@@ -7,34 +7,69 @@ import (
 type TableClient struct {
-	client *tableClient
-	cred   SharedKeyCredential
+	client  *tableClient
+	service *TableServiceClient
+	cred    SharedKeyCredential
+	name    string
 }
 
 // NewTableClient creates a TableClient object using the specified URL and request policy pipeline.
-func NewTableClient(serviceURL string, cred azcore.Credential, options *TableClientOptions) (*TableClient, error) {
-	con := newConnection(serviceURL, cred, options.getConnectionOptions())
+func NewTableClient(tableName string, serviceURL string, cred azcore.Credential, options *TableClientOptions) (*TableClient, error) {
+	s, err := NewTableServiceClient(serviceURL, cred, options)
+	return s.GetTableClient(tableName), err
+}
+
+func (t *TableClient) Name() string {
+	return t.name
+}
 
-	c, _ := cred.(*SharedKeyCredential)
+// Create creates the table with the name specified in NewTableClient
+func (t *TableClient) Create(ctx context.Context) (*TableResponseResponse, *runtime.ResponseError) {
+	return t.service.Create(ctx, t.name)
+}
+
+// Delete deletes the current table
+func (t *TableClient) Delete(ctx context.Context) (*TableDeleteResponse, *runtime.ResponseError) {
+	return t.service.Delete(ctx, t.name)
+}
 
-	return &TableClient{client: &tableClient{con}, cred: *c}, nil
+// Query queries the tables using the specified QueryOptions
+func (t *TableClient) Query(queryOptions QueryOptions) TableEntityQueryResponsePager {
+	return &tableEntityQueryResponsePager{tableClient: t, queryOptions: &queryOptions, tableQueryOptions: &TableQueryEntitiesOptions{}}
 }
 
-// Create
-func (t TableClient) Create(ctx context.Context, name string) (TableResponseResponse, error) {
-	resp, err := t.client.Create(ctx, TableProperties{&name}, nil, nil)
-	if resp == nil {
-		return TableResponseResponse{}, err
+func (t *TableClient) QueryAsStruct(opt QueryOptions, s FromMapper) StructEntityQueryResponsePager {
+	return &structQueryResponsePager{mapper: s, tableClient: t, queryOptions: &opt, tableQueryOptions: &TableQueryEntitiesOptions{}}
+}
+
+// AddMapEntity Creates an entity from a map value.
+func (t *TableClient) AddMapEntity(ctx context.Context, entity *map[string]interface{}) (*TableInsertEntityResponse, *runtime.ResponseError) {
+	toOdataAnnotatedDictionary(entity)
+	resp, err := t.client.InsertEntity(ctx, t.name, &TableInsertEntityOptions{TableEntityProperties: entity, ResponsePreference: ResponseFormatReturnNoContent.ToPtr()}, &QueryOptions{})
+	if err == nil {
+		insertResp := resp.(TableInsertEntityResponse)
+		return &insertResp, nil
 	} else {
-		return resp.(TableResponseResponse), err
+		return nil, convertErr(err)
 	}
 }
 
-// Delete
-func (t TableClient) Delete(ctx context.Context, name string) (TableDeleteResponse, error) {
-	return t.client.Delete(ctx, name, nil)
+// AddEntity creates an entity from an arbitrary struct value.
+func (t *TableClient) AddEntity(ctx context.Context, entity interface{}) (*TableInsertEntityResponse, *runtime.ResponseError) {
```

Comment:
This func is the analog for the QueryAsStruct method above. It allows adding of a "strongly typed" table entity struct rather than the default map shape.